sklearn/linear_model/base.py

```diff
@@ -91,6 +91,7 @@ def make_dataset(X, y, sample_weight, random_state=None):
                           seed=seed)
         intercept_decay = SPARSE_INTERCEPT_DECAY
     else:
+        X = np.ascontiguousarray(X)
         dataset = ArrayData(X, y, sample_weight, seed=seed)
```

Comment:
what if y isn't ?

Reply:
y with a 1D array should be both C and F array, isn't it.

Reply:
It can be non contiguous though `y[::2]`..

Reply:
Oh true.